core/src/main/java/org/elasticsearch/index/shard/IndexShard.java

```diff
@@ -1445,10 +1444,7 @@ void persistMetadata(ShardRouting newRouting, ShardRouting currentRouting) {
     }
 
     private String getIndexUUID() {
-        assert indexSettings.get(IndexMetaData.SETTING_INDEX_UUID) != null
-                || indexSettings.getAsVersion(IndexMetaData.SETTING_VERSION_CREATED, Version.CURRENT).before(Version.V_0_90_6) :
```

Comment:
I think this is from a time where SETTING_INDEX_UUID was introduced and we wanted to make sure we didn't forget to add it in some place. This is long gone and can be removed imo ...